src/main/openapi/discovery.yaml

```diff
@@ -209,15 +209,58 @@ components:
     DatasetSearchQueryFacet:
       type: object
       properties:
-        facetGroup:
+        source:
           type: string
-          title: Facet group
-        facet:
+          title: Facet source
+        type:
+          $ref: "#/components/schemas/FilterType"
+          title: Facet type
+        key:
+          type: string
+          title: Facet key
+        value:
+          type: string
+          title: Facet value
+        operator:
+          $ref: "#/components/schemas/Operator"
+          title: Operator
+        entries:
+          type: array
+          items:
+            $ref: '#/components/schemas/QueryEntry'
+      required:
+        - source
+        - type
+        - label
+    FilterType:
+      type: string
+      enum:
+        - DROPDOWN
+        - FREE_TEXT
+        - ENTRIES
+      title: Filter type
+    Operator:
+      type: string
+      enum:
+        - "="
+        - ">"
+        - "<"
+        - "!="
+        - "%"
+      x-enum-varnames:
+        - EQUALS
+        - GREATER_THAN
+        - LESS_THAN
+        - DIFFERENT
+        - CONTAINS
+    QueryEntry:
+      properties:
+        label:
```

Comment:
important: `FilterEntry` identifier is called `id`, so probably you should rename `QueryEntry.label` to `QueryEntry.id`, to avoid misunderstanding when combining the two endpoints.

Reply:
Good catch, I renamed their id to `key`